### PR TITLE
Update Colours page / Unify with atomic design guide

### PR DESCRIPTION
**Pull request by abenitoc**
Previously we had this page with some colour variables in Bridget:

image

Having a previous work of colours made by the team:

```
$cardinal-red: #bd0013;
$flamingo-red: #fb3f56;
$parrot-green: #37b083;
$london-grey:  #e9e9e9;
$pigeon-grey:  #808080;
$catbird-grey: #cccccc;
$finch-blue:   #3a4456;

```

And some others proposed by Alex in the Atomic Design guide he is working on:

```
$inactive-gray: #d5d5d5;

$dark-gray-1: #3d3d3d;
$dark-gray-2: #686c71;
$dark-gray-3: #84898f;
$light-gray-1: #b2bac4;
$light-gray-2: #f8f8f8;

$icon-resting: #263753;

```

I've updated the colours/variable page with the colours got from the atomic design with the following shape:

image

Also is worth to mention if is it worth to add some of them or which of them as `theme-colours` and `colours` in the mapping file in Bridget.

⚠️ This is a WIP for discussion, not intended to be merged as is, but I think a lot of feedback is appreciated. Colours by Alex are still refining so until I can confirm this, I'll standby it.

### CODE REVIEW

**Comment by doutatsu** [comment]
@abenitoc Can we actually use the same approach as Alex: put these colours into cards. Its nicer on the eye and some of the colours would look better inside a card: for example, white is exactly the same as the background, so it looks weird :(

this looks a lot better:
image

File: src/scss/bootstrap/_variables.scss [hunk 1]
```diff
@@ -5,7 +5,7 @@
 
 // // Color system
 
-// $white:    #fff !default;
+$white:    #fff !default;
```
Comment (doutatsu) [inline src/scss/bootstrap/_variables.scss]: We shouldn't uncomment if we are not overwriting these. If they are commented out, it just means we are using the default, which we are

File: src/scss/bootstrap/_variables.scss [hunk 2]
```diff
@@ -15,7 +15,7 @@
 // $gray-700: #495057 !default;
 // $gray-800: #343a40 !default;
 // $gray-900: #212529 !default;
-// $black:    #000 !default;
+$black:    #000 !default;
```
Comment (doutatsu) [inline src/scss/bootstrap/_variables.scss]: Same as above

File: src/scss/bootstrap/_variables.scss [hunk 3]
```diff
@@ -45,11 +45,11 @@
 // $teal:    #20c997 !default;
 // $cyan:    #17a2b8 !default;
 
-// $colors: () !default;
+ $colors: () !default;
 // // stylelint-disable-next-line scss/dollar-variable-default
-// $colors: map-merge(
-//   (
-//     "blue":       $blue,
+$colors: map-merge(
```
Comment (doutatsu) [inline src/scss/bootstrap/_variables.scss]: As far as I can tell, we don't need to uncomment this either. You removed `blue` for `black` (why), but didn't uncomment, so this didn't go into effect anyways...

File: src/scss/bootstrap/_variables.scss [hunk 4]
```diff
@@ -62,34 +62,45 @@
...
-// $primary:       $blue !default;
-$secondary:     $pigeon-grey;
+$primary:       $finch-blue;
+$secondary:     $cardinal-red;
 $success:       $parrot-green;
 // $info:          $cyan !default;
 // $warning:       $yellow !default;
 $danger:        $flamingo-red;
 // $light:         $gray-100 !default;
+$inactive: $inactive-gray;
```
Comment (doutatsu) [inline src/scss/bootstrap/_variables.scss]: We shouldn't be adding new variables into here. I don't see `$inactive` as part of Bootstrap variable, so we should keep this utility variable in our own overrides

**Comment by abenitoc** [comment]
Yes definitely, I like it better that way.
I had in my head something similar, to copy Alex design with a bit of 'hydration'. I had in my head something like:

_______________
|                      |
|                      |
|                      |
--------------
| Hex color     |
|css variable  |
|human name|
--------------

I think is valuable to have the reference of the css variable but other than that I agree 100% with you, also have separated by different utilities, and maybe include some description if it is primary secondary?.

File: src/scss/bootstrap/_variables.scss [hunk 5]
```diff
@@ -62,34 +62,45 @@
...
 $dark:          $finch-blue;
 
-// $theme-colors: () !default;
+$dark-gray-1: $dark-gray-1;
+$dark-gray-2: $dark-gray-2;
+$dark-gray-3: $dark-gray-3;
+
+ $theme-colors: () !default;
```
Comment (doutatsu) [inline src/scss/bootstrap/_variables.scss]: Again, you are not overriding the existing Bootstrap variables but instead adding completely new ones, which I don't think we should be doing

File: src/scss/bootstrap/_variables.scss (hunk 1 above)
Comment (abenitoc) [reply src/scss/bootstrap/_variables.scss]: image

If I uncomment any of those lines I don't have access to use bg-black later, I'm okay with not doing it, but we need to manage to override for background otherwise, or maybe have custom classes just for bridget, any ideas?

File: src/scss/bootstrap/_variables.scss (hunk 2 above)
Comment (abenitoc) [reply src/scss/bootstrap/_variables.scss]: ⬆️

File: src/scss/bootstrap/_variables.scss (hunk 5 above)
Comment (abenitoc) [reply src/scss/bootstrap/_variables.scss]: In some cases we should not, i.e. we might want to have the grays as themselves, while using $primary maybe make sense to just have cardinal-red as the primary colour.

File: src/scss/bootstrap/_variables.scss (hunk 4 above)
Comment (abenitoc) [reply src/scss/bootstrap/_variables.scss]: This is true, but I haven't found exactly a good fit for $inactive. Any suggestions?

**Comment by gnclmorais** [comment]
I’ll look into this PR this afternoon 🕵🏻‍♂️

File: src/scss/bootstrap/_variables.scss (hunk 1 above)
Comment (doutatsu) [reply src/scss/bootstrap/_variables.scss]: That's weird, that should not be the case. Will take a look into our scss imports

File: src/scss/bootstrap/_variables.scss [hunk 6]
```diff
@@ -62,34 +62,45 @@
...
+     "light-gray-1": $light-gray-1,
+     "light-gray-2": $light-gray-2,
+     "dark-gray-1": $dark-gray-1,
+     "dark-gray-2": $dark-gray-2,
+     "dark-gray-3": $dark-gray-3,
+     "white":       $white,
+     "black":       $black
+   ),
+   $theme-colors
+ );
```
Comment (gnclmorais) [inline src/scss/bootstrap/_variables.scss]: I’m not sure if you are all familiar with the [`$theme-colors` map][theme-colors-map], but let me tell you why I don’t think we should add anything to it (at least for now).

☙

Berto, did you see the CSS generated by this change? Let me show you part of it, the bit generated for `light-gray-2`:
```
Searching 1 file for "light-gray-2" (case sensitive)

/Users/goncalomorais/Development/BridgeU/_repos/bridget/dist/bridget.css:
   40    --inactive: #d5d5d5;
   41    --light-gray-1: #b2bac4;
   42:   --light-gray-2: #f8f8f8;
   43    --dark-gray-1: #3d3d3d;
   44    --dark-gray-2: #686c71;
   ..
 1260      background-color: #dbe0e3; }
 1261  
 1262: .table-light-gray-2,
 1263: .table-light-gray-2 > th,
 1264: .table-light-gray-2 > td {
 1265    background-color: #fdfdfd; }
 1266  
 1267: .table-light-gray-2 th,
 1268: .table-light-gray-2 td,
 1269: .table-light-gray-2 thead th,
 1270: .table-light-gray-2 tbody + tbody {
 1271    border-color: #fbfbfb; }
 1272  
 1273: .table-hover .table-light-gray-2:hover {
 1274    background-color: #f0f0f0; }
 1275:   .table-hover .table-light-gray-2:hover > td,
 1276:   .table-hover .table-light-gray-2:hover > th {
 1277      background-color: #f0f0f0; }
 1278  
 ....
 2080        box-shadow: 0 0 0 0.2rem rgba(156, 164, 173, 0.5); }
 2081  
 2082: .btn-light-gray-2 {
 2083    color: #212529;
 2084    background-color: #f8f8f8;
 2085    border-color: #f8f8f8; }
 2086:   .btn-light-gray-2:hover {
 2087      color: #212529;
 2088      background-color: #e5e5e5;
 2089      border-color: #dfdfdf; }
 2090:   .btn-light-gray-2:focus, .btn-light-gray-2.focus {
 2091      box-shadow: 0 0 0 0.2rem rgba(216, 216, 217, 0.5); }
 2092:   .btn-light-gray-2.disabled, .btn-light-gray-2:disabled {
 2093      color: #212529;
 2094      background-color: #f8f8f8;
 2095      border-color: #f8f8f8; }
 2096:   .btn-light-gray-2:not(:disabled):not(.disabled):active, .btn-light-gray-2:not(:disabled):not(.disabled).active,
 2097:   .show > .btn-light-gray-2.dropdown-toggle {
 2098      color: #212529;
 2099      background-color: #dfdfdf;
 2100      border-color: #d8d8d8; }
 2101:     .btn-light-gray-2:not(:disabled):not(.disabled):active:focus, .btn-light-gray-2:not(:disabled):not(.disabled).active:focus,
 2102:     .show > .btn-light-gray-2.dropdown-toggle:focus {
 2103        box-shadow: 0 0 0 0.2rem rgba(216, 216, 217, 0.5); }
 2104  
 ....
 2428        box-shadow: 0 0 0 0.2rem rgba(178, 186, 196, 0.5); }
 2429  
 2430: .btn-outline-light-gray-2 {
 2431    color: #f8f8f8;
 2432    border-color: #f8f8f8; }
 2433:   .btn-outline-light-gray-2:hover {
 2434      color: #212529;
 2435      background-color: #f8f8f8;
 2436      border-color: #f8f8f8; }
 2437:   .btn-outline-light-gray-2:focus, .btn-outline-light-gray-2.focus {
 2438      box-shadow: 0 0 0 0.2rem rgba(248, 248, 248, 0.5); }
 2439:   .btn-outline-light-gray-2.disabled, .btn-outline-light-gray-2:disabled {
 2440      color: #f8f8f8;
 2441      background-color: transparent; }
 2442:   .btn-outline-light-gray-2:not(:disabled):not(.disabled):active, .btn-outline-light-gray-2:not(:disabled):not(.disabled).active,
 2443:   .show > .btn-outline-light-gray-2.dropdown-toggle {
 2444      color: #212529;
 2445      background-color: #f8f8f8;
 2446      border-color: #f8f8f8; }
 2447:     .btn-outline-light-gray-2:not(:disabled):not(.disabled):active:focus, .btn-outline-light-gray-2:not(:disabled):not(.disabled).active:focus,
 2448:     .show > .btn-outline-light-gray-2.dropdown-toggle:focus {
 2449        box-shadow: 0 0 0 0.2rem rgba(248, 248, 248, 0.5); }
 2450  
 ....
 4087      box-shadow: 0 0 0 0.2rem rgba(178, 186, 196, 0.5); }
 4088  
 4089: .badge-light-gray-2 {
 4090    color: #212529;
 4091    background-color: #f8f8f8; }
 4092:   a.badge-light-gray-2:hover, a.badge-light-gray-2:focus {
 4093      color: #212529;
 4094      background-color: #dfdfdf; }
 4095:   a.badge-light-gray-2:focus, a.badge-light-gray-2.focus {
 4096      outline: 0;
 4097      box-shadow: 0 0 0 0.2rem rgba(248, 248, 248, 0.5); }
 ....
 4273      color: #45484b; }
 4274  
 4275: .alert-light-gray-2 {
 4276    color: #818181;
 4277    background-color: #fefefe;
 4278    border-color: #fdfdfd; }
 4279:   .alert-light-gray-2 hr {
 4280      border-top-color: #f0f0f0; }
 4281:   .alert-light-gray-2 .alert-link {
 4282      color: #686868; }
 4283  
 ....
 4617      border-color: #5d6166; }
 4618  
 4619: .list-group-item-light-gray-2 {
 4620    color: #818181;
 4621    background-color: #fdfdfd; }
 4622:   .list-group-item-light-gray-2.list-group-item-action:hover, .list-group-item-light-gray-2.list-group-item-action:focus {
 4623      color: #818181;
 4624      background-color: #f0f0f0; }
 4625:   .list-group-item-light-gray-2.list-group-item-action.active {
 4626      color: #fff;
 4627      background-color: #818181;
 ....
 5381    background-color: #95a0ae !important; }
 5382  
 5383: .bg-light-gray-2 {
 5384    background-color: #f8f8f8 !important; }
 5385  
 5386: a.bg-light-gray-2:hover, a.bg-light-gray-2:focus,
 5387: button.bg-light-gray-2:hover,
 5388: button.bg-light-gray-2:focus {
 5389    background-color: #dfdfdf !important; }
 5390  
 ....
 5495    border-color: #b2bac4 !important; }
 5496  
 5497: .border-light-gray-2 {
 5498    border-color: #f8f8f8 !important; }
 5499  
 ....
 7775    color: #8793a3 !important; }
 7776  
 7777: .text-light-gray-2 {
 7778    color: #f8f8f8 !important; }
 7779  
 7780: a.text-light-gray-2:hover, a.text-light-gray-2:focus {
 7781    color: #d2d2d2 !important; }
 7782  

56 matches in 1 file
```

An extra variable on the `$theme-colours` map creates an additional **56 [CSS selectors][css-selectors]**. There are eight new values on this map, so multiply that… The CSS file grew from `174 KiB` to `202 KiB` (Webpack even raised a warning about its size).

This happened because Bootstrap generates coloured “assets” for each of those values: buttons, text utilities, list groups, tables, etc.… That’s not what we want, at least not now. We want to add these variables as simple atoms for the moment, until they get semantic meaning.

☙

Related to this, please don’t translate directly from the InVision prototypes. I can’t stress this enough, but we are supposed to double-check the designs we get. That means:
- Know quite well how Bootstrap works;
- Understand the meaning of the design;
- Figure out if there are missing pieces on the prototypes.

For example:
image
What does this mean? 70%… what, opacity? And _Icon Button (Resting)_, what does that even mean?

Another example: 
image
So anything that is _inactive_ will have this colour? Does _inactive_ mean _disabled_?

☙

I hope this makes sense, we need to be **highly** critical of the design before implementing it. It will pay off.


[theme-colors-map]: https://getbootstrap.com/docs/4.3/getting-started/theming/#maps-and-loops
[css-selectors]: https://developer.mozilla.org/en-US/docs/Learn/CSS/Introduction_to_CSS/Selectors

File: src/scss/bridgeu/_variables.scss [hunk 7]
```diff
@@ -19,6 +19,15 @@ $pigeon-grey:  #808080;
 $catbird-grey: #cccccc;
 $finch-blue:   #3a4456;
 
+$inactive-gray: #d5d5d5;
```
Comment (gnclmorais) [inline src/scss/bridgeu/_variables.scss]: I would like to get more information about this variable/colour before adding it.

File: src/scss/bridgeu/_variables.scss [hunk 8]
```diff
@@ -19,6 +19,15 @@ $pigeon-grey:  #808080;
...
+
+$dark-gray-1: #3d3d3d;
+$dark-gray-2: #686c71;
+$dark-gray-3: #84898f;
+$light-gray-1: #b2bac4;
+$light-gray-2: #f8f8f8;
+
+$icon-resting: #263753;
```
Comment (gnclmorais) [inline src/scss/bridgeu/_variables.scss]: I would like to get more information about this variable/colour before adding it.

**Comment by abenitoc** [comment]
Thanks for your comments, here.

This was not intended to be merged, but it was a raw state to grow this further. This is still a Work In progress and not intended to be reviewed yet or shown as something like that.

Yesterday Max reviewed and asked for changes but I did not ask for review, it is not ready, I wanted to talk to Gonçalo and have this reference to talk about this at some point with Alex not to leave it as is and talk about refinement slowly. Having said so...

Yes, it does not make sense and in general we previously talked about merging things in Bridget and I have notions of the overrides. I knew it was not the right approach but the right approach to have a fast place to comment.

Long story short, thanks a lot for the long comment @gnclmorais. It is a good point of reference to anyone that writes Bridget from now on.

Maybe I was not clear with my intentions in this PR by answering @doutatsu briefly and without context.
Whenever we have more conversations on this, I'll take responsibility on updating this PR.
Hopefully we can talk first on many things done so far and align my intentions, with Alex point of view and yours Gonçalo, Max and Shri to achieve best Bridget possible. 😄

**Comment by doutatsu** [comment]
@abenitoc Sorry, I knew its just a WIP, but left a "review". Should have just commented instead, I wanted to raise a flag in the approach, which is why I used `request changes`

I know Gonçalo had specific thinking around how to override Bootstrap, as I hit that and Gonçalo explained it to me in detail, so wanted to make sure that knowledge keeps spreading, as that will be needed if everyone, not just Gonçalo, will be able to merge code here. Perhaps, writing a wiki for this would be a good thing to start

**Comment by abenitoc** [comment]
Close as this is an old updated PR
